Travel Scheduler/View Controllers/HomeCollectionViewController.m

```diff
@@ -107,15 +103,59 @@ -(void) createButtonToMenu
     [self.buttonToMenu addTarget: self action: @selector(buttonClicked:) forControlEvents: UIControlEventTouchUpInside];
 }
 
-- (void) buttonClicked: (id)sender
+- (void)buttonClicked:(id)sender
 {
     [self animateView];
     [self.leftViewToSlideIn createButtonToCloseSlideIn];
 }
 
+- (void)makeCloseButton
+{
+    UIBarButtonItem *item = [[UIBarButtonItem alloc] initWithTitle:@"Back" style:UIBarButtonItemStylePlain target:self action:@selector(returnToFirstScreen:)];
+    [item setTitleTextAttributes:@{
+                                   NSFontAttributeName: [UIFont fontWithName:@"Gotham-Light" size:17.0],
+                                   NSForegroundColorAttributeName: [UIColor blackColor]
+                                   } forState:UIControlStateNormal];
+    [self.navigationItem setLeftBarButtonItem:item animated:YES];
+}
+    
+- (void)setStateOfCreateScheduleButton
+{
+    if((self.arrayOfSelectedPlaces.count == 0 || self.isScheduleUpToDate) && !self.scheduleButton.hidden) {
+        [self animateScheduleButton];
+    } else if(!self.isScheduleUpToDate && self.hasFirstSchedule) {
+        [self.scheduleButton setTitle:@"Regenerate Schedule" forState:UIControlStateNormal];
+        if(self.scheduleButton.hidden) {
+            [self animateScheduleButton];
+        }
+    } else {
+        [self.scheduleButton setTitle:@"Generate Schedule" forState:UIControlStateNormal];
+        if(self.scheduleButton.hidden) {
+            [self animateScheduleButton];
+        }
+    }
+        
+}
+ 
+- (void)animateScheduleButton
+{
+    if(self.scheduleButton.isHidden) {
```

Comment:
Since you check if the schedule button is hidden before you call the method and also inside the method again, it seems a little repetitive. Not a big issue, but maybe you could consider splitting the animateScheduleButton to animateScheduleButtonIn and animateScheduleButtonOut so you would only need to check the if statement once.